enginetest/queries/queries.go

```diff
@@ -1666,6 +1666,14 @@ var QueryTests = []QueryTest{
 		Query:    "SELECT 1 IN (2,3,4)",
 		Expected: []sql.Row{{false}},
 	},
+	{
+		Query:    "SELECT 1 FROM DUAL WHERE 1 IN (SELECT '1' FROM DUAL)",
+		Expected: []sql.Row{{1}},
+	},
+	{
+		Query:    "SELECT '1' FROM DUAL WHERE 1 IN (SELECT '1' FROM DUAL)",
```

Comment:
I'd change this one to: 
```suggestion
		Query:    "SELECT 1 FROM DUAL WHERE '1' IN (SELECT '1' FROM DUAL)",
```

otherwise, you're essentially testing the same SubqueryIn code path twice (since it's converting and comparing to an int in both cases)